sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsQueryResult.java

```diff
@@ -15,17 +15,21 @@
     private final List<LogsTable> logsTables;
     private final LogsQueryStatistics statistics;
     private final LogsQueryError error;
+    private final LogsQueryVisualization visualization;
 
     /**
      * Creates an instance {@link LogsQueryResult} with a list of {@link LogsTable}.
      * @param logsTables The list of {@link LogsTable} returned as query result.
      * @param statistics The query execution statistics.
      * @param error The error details if there was an error executing the query.
+     * @param visualization The visualization information for the logs query.
      */
-    public LogsQueryResult(List<LogsTable> logsTables, LogsQueryStatistics statistics, LogsQueryError error) {
+    public LogsQueryResult(List<LogsTable> logsTables, LogsQueryStatistics statistics,
+                           LogsQueryVisualization visualization, LogsQueryError error) {
```

Comment:
how is the viaulation getting used? Is the service now doing it themselves?

Reply:
> how is the viaulation getting used? 

"visualization" is just a json that users can deserialize if they want to. It's a fairly niche feature and the swagger does not define a model structure. So, we just return the raw data. 

> Is the service now doing it themselves?

This is a response type. The service returns this data to the user.